lib/src/intl_suggestors/intl_importer.dart

```diff
@@ -38,27 +40,48 @@ Stream<Patch> intlImporter(
 
   if (!needsIntlImport) return;
 
-  final intlUri = 'package:${projectName}/src/intl/${projectName}_intl.dart';
+  final currentFilePath = Platform.script.toFilePath();
+  final currentDirectory = path.dirname(currentFilePath);
+
+  final intlUri = 'package:$projectName/src/intl/${projectName}_intl.dart';
+  final relativePathToIntlDir = path.relative(currentDirectory, from: intlUri);
+
+  final levelsUp = List<String>.generate(relativePathToIntlDir.split(path.separator).length, (_) => '..').join('/');
+
+  final relativeImportPath = '$levelsUp/src/intl/${projectName}_intl.dart';
 
   final insertInfo = _insertionLocationForPackageImport(
-      intlUri, mainLibraryUnitResult.unit, mainLibraryUnitResult.lineInfo);
+      relativeImportPath, mainLibraryUnitResult.unit, mainLibraryUnitResult.lineInfo);
```

Comment:
How does this fix the ordering problem? It seems like this is always finding the insert position as if the import path was relative. Shouldn't you compute the insertInfo after you have the importStatement?